parser-typechecker/src/Unison/Lexer.hs

```diff
@@ -60,74 +97,440 @@ type IsVirtual = Bool -- is it a virtual semi or an actual semi?
 
 makePrisms ''Lexeme
 
+space :: P ()
+space = LP.space CP.space1 (fold <|> LP.skipLineComment "--")
+                           (LP.skipBlockCommentNested "{-" "-}")
+  where
+  fold = P.try $ lit "---" *> P.takeRest *> pure ()
+
+lit :: String -> P String
+lit = P.try . LP.symbol (pure ())
```

Comment:
Just curious, did you find the `try` here was necessary?

I had thought that e.g. `symbol "foo"` would not advance the parser state unless all of "f", "o", and "o" were matched.

Reply:
I wasn't actually sure but I kinda like it better with the `try`, since you could certainly imagine a version of `symbol` that didn't `try`.